### PR TITLE
JC-2273 Map files are generated when JS compression happens

### PR DESCRIPTION
**Pull request by MuH3gPaB**
Wro4j doesn't support SourceMaps, in this case another stuff was used.
Now js and css files are compressed by primefaces extensions
resources optimizer maven plugin.
Sourcemaps are generated at the same moment.

All configuration in jcommunue-web-view/pom.xml
wro.xml is not necessary any more.

Next step: Reorganize resources folder, so we could  make pom.xml more compact.

### CODE REVIEW

**Comment by ctapobep** [comment]
Looks good at first glance. From what I understood you kept the same paths/filenames but just used a different plugin, right? 

@masyan, what do you think?

**Comment by ctapobep** [comment]
Did you check that there are no warnings/errors in the JS console?

**Comment by masyan** [comment]
'Wro4j doesn't support SourceMaps' you mean wildcard? wro4j supprt id (http://wro4j.readthedocs.io/en/stable/WildcardSupport/). So, size will not decrease.

**Comment by MuH3gPaB** [comment]
> you kept the same paths/filenames but just used a different plugin, right?

yes, you right
i've used another plugin, that supports generating source maps
all paths/filenames don't changed
I don't think paths are good for this config, cause we have to aggregate 
files from different subfolders, so we have to include each file separately
but it's not a problem, everything work fine

> Did you check that there are no warnings/errors in the JS console?

I think there was no errors or warnings, I'll check again tomorrow

> 'Wro4j doesn't support SourceMaps' you mean wildcard?

Not right. SourceMaps used to restore original sources from compressed js

**Comment by MuH3gPaB** [comment]
> Did you check that there are no warnings/errors in the JS console?

In JS console are two warnings same as at javatalks.ru

Lots of warnings at CSS console, and they are same as at javatalks.ru too.

**Comment by ctapobep** [comment]
Awesome. Will merge this when the new spring starts.

File: jcommune-view/jcommune-web-view/pom.xml [hunk 1]
```diff
@@ -140,35 +140,290 @@
...
+                <!--LIBS - CSS-->
+                <include>/resources/css/lib/jquery.contextMenu.css</include>
+                <include>/resources/css/lib/jquery-ui.css</include>
+                <include>/resources/css/lib/chosen.css</include>
+              </includes>
+
+              <aggregations>
+                <aggregation>
+                  <outputFile>
+                    ${basedir}/src/main/webapp/resources/wro/main.css
```
Comment (masyan) [inline jcommune-view/jcommune-web-view/pom.xml]: remove "wro"

File: jcommune-view/jcommune-web-view/pom.xml [hunk 2]
```diff
@@ -140,35 +140,290 @@
...
+                <include>resources/javascript/app/events.js</include>
+                <include>resources/javascript/app/banner.js</include>
+                <include>resources/javascript/app/forumAdministration.js</include>
+                <include>resources/javascript/app/editPermissions.js</include>
+              </includes>
+
+              <aggregations>
+                <aggregation>
+                  <outputFile>
+                    ${basedir}/src/main/webapp/resources/wro/main.js
```
Comment (masyan) [inline jcommune-view/jcommune-web-view/pom.xml]: wro

File: jcommune-view/jcommune-web-view/pom.xml [hunk 3]
```diff
@@ -140,35 +140,290 @@
...
+                <include>resources/javascript/app/contextMenu.js</include>
+                <include>resources/javascript/lib/purl.js</include>
+                <include>resources/javascript/app/pollPreview.js</include>
+                <include>resources/javascript/app/codeHighlighting.js</include>
+              </includes>
+
+              <aggregations>
+                <aggregation>
+                  <outputFile>
+                    ${basedir}/src/main/webapp/resources/wro/pm.js
```
Comment (masyan) [inline jcommune-view/jcommune-web-view/pom.xml]: etc.

**Comment by ctapobep** [comment]
@MuH3gPaB could you please squash everything into 1 commit? 

Also, if it's not hard could you please try running system functional tests (via selenium) locally (since there are a lot of changes that potentially could've broken any place in JC): https://github.com/jtalks-org/functional-tests/tree/master/functional-tests-jcommune

They take a lot of time (an hour or two), so it's worth to leave them running when you go to bed :)

**Comment by MuH3gPaB** [comment]
Ok. I'll do it.

**Comment by MuH3gPaB** [comment]
Commits squashed, tests passed =) everything looks fine.

**Comment by ctapobep** [comment]
Merged!

**Pull request closed without merging**